src/Raven.Server/Storage/Schema/Updates/Documents/From10.cs

```diff
@@ -0,0 +1,142 @@
+                        continue;
+
+                    var writeTable = step.WriteTx.OpenTable(TombstonesSchema, tableName);
+                    // We seek by an index instead the PK because 
+                    // we weed to ensure that we aren't accessing an IsGlobal key
+                    foreach (var read in readTable.SeekForwardFrom(TombstonesSchema.FixedSizeIndexes[CollectionEtagsSlice], 0, 0))
+                    {
+                        CopyReadMemory(context, read);
+
+                        var type = *(DocumentTombstone.TombstoneType*)read.Reader.Read((int)TombstoneTable.Type, out _);
+                        var oldCollection = TableValueToString(context, (int)TombstoneTable.Collection, ref read.Reader);
+                        using (DocumentIdWorker.GetStringPreserveCase(context, oldCollection, out Slice collectionSlice))
+                        using (writeTable.Allocate(out TableValueBuilder write))
+                        {
+                            write.Add(read.Reader.Read((int)TombstoneTable.LowerId, out int size), size);
+                            write.Add(read.Reader.Read((int)TombstoneTable.Etag, out size), size);
+                            write.Add(read.Reader.Read((int)TombstoneTable.DeletedEtag, out size), size);
+                            write.Add(read.Reader.Read((int)TombstoneTable.TransactionMarker, out size), size);
+                            write.Add(read.Reader.Read((int)TombstoneTable.Type, out size), size);
+                            if (type == DocumentTombstone.TombstoneType.Attachment)
+                            {
+                                write.Add(read.Reader.Read((int)TombstoneTable.Collection, out size), size);
+                            }
+                            else
+                            {
+                                write.Add(collectionSlice);
+                            }
+                            write.Add(read.Reader.Read((int)TombstoneTable.Flags, out size), size);
+                            write.Add(read.Reader.Read((int)TombstoneTable.ChangeVector, out size), size);
+                            write.Add(read.Reader.Read((int)TombstoneTable.LastModified, out size), size);
+                            writeTable.Set(write);
+                        }
+                    }
+                }
+            }
+
+            // Update conflicts' collection value
+            using (step.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var readTable = step.ReadTx.OpenTable(ConflictsSchema, ConflictsSlice);
+                if (readTable != null)
+                {
+                    var writeTable = step.WriteTx.OpenTable(ConflictsSchema, ConflictsSlice);
+                    foreach (var read in readTable.SeekByPrimaryKey(Slices.BeforeAllKeys, 0))
+                    {
+                        CopyReadMemory(context, read);
```

Comment:
This leaks memory, we only need the memory for a short duration, and we need to release it immediately.
The context will recover this, of course, but for large system, we want to do this cleanly

Reply:
Right, I'll fix this.

Reply:
https://github.com/ravendb/ravendb/pull/4035